### PR TITLE
feat(llm_connection): add openrouter api endpoint

### PR DESCRIPTION
**Pull request by mrgoonie**
I added [OpenRouter API](https://openrouter.ai/) endpoint so we can use other LLM models (such as Google Bard, Claude,...)

### CODE REVIEW

File: pilot-env/pyvenv.cfg
Comment (LeonOstrez) [inline pilot-env/pyvenv.cfg:1]: how come you commited whole pilot-env folder?

File: pilot/.env.example [hunk 1]
```diff
@@ -1,13 +1,24 @@
-#OPENAI or AZURE
-ENDPOINT=OPENAI
+# OPENAI or AZURE or OPENROUTER
+ENDPOINT=OPENROUTER
```
Comment (LeonOstrez) [inline pilot/.env.example:2]: let's try to keep all defaults same as before. So in this case ENDPOINT=OPENAI

File: pilot/.env.example [hunk 2]
```diff
@@ -1,13 +1,24 @@
...
+
+OPENAI_ENDPOINT=
 OPENAI_API_KEY=
+
 AZURE_API_KEY=
 AZURE_ENDPOINT=
-#In case of Azure endpoint, change this to your deployed model name
-MODEL_NAME=gpt-4
```
Comment (LeonOstrez) [inline pilot/.env.example:7]: same as above, lets keep default to gpt-4

File: pilot/.env.example [hunk 3]
```diff
@@ -1,13 +1,24 @@
...
+
+OPENROUTER_API_KEY=
+OPENROUTER_ENDPOINT=https://openrouter.ai/api/v1/chat/completions
+
+# In case of Azure/OpenRouter endpoint, change this to your deployed model name
+MODEL_NAME=openai/gpt-3.5-turbo-16k
 MAX_TOKENS=8192
+
+# Database
+DATABASE_TYPE=postgres
```
Comment (LeonOstrez) [inline pilot/.env.example:18]: same as above lets keep db sqlite

File: pilot/.env.example (hunk 1 above)
Comment (mrgoonie) [reply pilot/.env.example]: Done

File: pilot-env/pyvenv.cfg
Comment (mrgoonie) [reply pilot-env/pyvenv.cfg:1]: Added to `.gitignore` already

**Review by LeonOstrez — CHANGES_REQUESTED** [review]
can you please remove whole pilot-env folder from pull request and then i can merge it. Thanks for making PR!

**Comment by mrgoonie** [comment]
> I suppose support for Code Llama would be out of scope for this PR...

You mean this one?
image

OpenRouter supports many LLMs, you can check out at their website: https://openrouter.ai/docs

**Pull request merged**